### ISSUE TITLE
Support Prometheus

### ISSUE THREAD

**Issue by benjaminguttmann-avtq** [Vonage/Grafana_Status_panel#17]
Add support for Prometheus data sources. 
I would really appreciate it to use this add-on with prometheus data source.

**Comment by jnials** [comment]
Yep.  This would solve an immediate problem for me.

**Comment by jangor** [comment]
How can we help? I would like to give it a try but don't know how to setup the development environment. What transpiler to use?

**Comment by Kumaravel2904** [comment]
Yes, this would be a useful enhancement

**Comment by mrngm** [comment]
I found out that Prometheus as data source works out of the box when you set the `Legend format` of the query in the Metrics panel the same as the `Alias` belonging to that metric in the Options panel.

Although I'm not quite sure why this is required, I think it has to do with [this line](https://github.com/Vonage/Grafana_Status_panel/blob/master/src/status_ctrl.js#L92). I set a breakpoint on line 95 and tried some various names a bit and found above workaround.

**Comment by TomerAmirV** [comment]
Hi @mrngm , Thanks for your input! I hope it will help more people.
The reason why not all data sources are supported is that Grafana does not guarantee that the order in which the series are returned is the order they were made (the order you wrote them). In the plugin, we need to know the order so we can check the thresholds for the data and make some calculation (like the aggregations) according to the setting you can make.

Unfortunately, every data source has its own convention on how the data is returned and identified, so we do not have a generic solution for it.

I contacted the Grafana developers, and they told me that this is a known issue and that they are working on it.

As we have other projects we're working on and we do not use Prometheus, I don't know if/when we'll have time to add support for it.

You could wait until Grafana addresses the issue I told you about, or you are welcome to fix it and make a pull request. We'll be happy to merge it once we check it. 😄 

I'll be happy to help with any questions you might have when adding the support. I looked at the line you mentioned, and I think that the code was changed since you marked it, since now it shows an empty line.

**Comment by mrngm** [comment]
> I looked at the line you mentioned, and I think that the code was changed since you marked it, since now it shows an empty line.

Yep, it has, [this commit](https://github.com/Vonage/Grafana_Status_panel/blob/4ac1c80872591f29af00224a258bec0cfeb2d7c6/src/status_ctrl.js#L92) still has the line I wanted to hilight.

Nevertheless, I'm not (yet) experienced enough with either Grafana or Prometheus to _really_ find out where and how to apply patches.

**Comment by geraldcor** [comment]
So, this isn't necessarily a bad thing to have to set Alias and Legend format the same. It's weird, but it works. Could be just a documentation thing?

**Comment by wrouesnel** [comment]
The way this works isn't super-conducive to Prometheus at the moment. If you have a lot of servers (i.e. like my 150), you wind up templating out a dashboard which sends way too many individual queries per update.

What you want is to be able to have multiple "status panels" in one panel track the series returned from a single query, and then populate based on labels.

**Comment by benjaminguttmann-avtq** [comment]
I got the following metric in my Prometheus 

`bosh_job_process_healthy{bosh_deployment="cf-5b3b1746c163508ed360",bosh_job_name="mysql",bosh_job_process_name="streaming-mysql-backup-tool",job="bosh"}` 

do somebody know if it is possible to use the value of the bosh_job_process_name as an alias. So that several processes inside the job gets named different inside one query. I hope it is clear what I am trying to do. Any ideas? Suggestions?

**Comment by keithf4** [comment]
Putting in another request for prometheus support.

**Comment by andrepoleza** [comment]
Hello :) Any updates on this?
